assembl/static2/js/app/pages/idea.jsx

```diff
@@ -103,6 +104,17 @@ export const transformPosts = (edges, messageColumns, additionnalProps = {}) =>
     .sort(creationDateLastDescendantComparator);
 };
 
+// Function that counts the total number of posts in a Bright Mirror debate section
+// transformedFilteredPosts parameter is built from transformPosts filtered with the current displayed fiction
+export const getDebateTotalMessages = (transformedFilteredPosts: [Object]) => {
+  if (transformedFilteredPosts.length) {
+    return (
+      1 + getDebateTotalMessages(transformedFilteredPosts[0].children) + getDebateTotalMessages(transformedFilteredPosts.slice(1))
```

Comment:
What is the "1" corresponding to here ?

Reply:
It's an increment in a recursive function. You add one on each leaf and go down by one level on each child.